File: test-api.js
import { ChatGoogleGenerativeAI } from "@langchain/google-genai";
import dotenv from "dotenv";
import { RollDTool } from "./dist/tools/dice-tools.js";
import { ChatPromptTemplate } from "@langchain/core/prompts";
import { AgentExecutor, createToolCallingAgent } from "langchain/agents";
import { setGlobalDispatcher, ProxyAgent } from "undici";
// 只有在设置了代理的情况下才使用代理
if (process.env.https_proxy) {
  const dispatcher = new ProxyAgent({ uri: new URL(process.env.https_proxy).toString() });
  setGlobalDispatcher(dispatcher);
}
dotenv.config();

async function testAPI() {
  console.log("=== Google Gemini API 测试 ===");
  console.log("API Key:", process.env.GOOGLE_API_KEY ? "已设置" : "未设置");
  console.log("Model:", process.env.GEMINI_MODEL || "gemini-pro");
  
  if (!process.env.GOOGLE_API_KEY) {
    console.error("❌ 错误：未设置 GOOGLE_API_KEY");
    console.log("请检查 .env 文件");
    return;
  }

  try {
    const llm = new ChatGoogleGenerativeAI({
      model: "gemini-2.0-flash",
      apiKey: process.env.GOOGLE_API_KEY,
      temperature: 0.1,
    });

    const tools = [RollDTool];
    const prompt = ChatPromptTemplate.fromTemplate(`你是一位经验丰富的《克苏鲁的呼唤》守秘人(KP)。
      战斗时，需要先扔骰子判定命中，再判定对方的闪避或对抗。如果确实命中再投伤害骰子判定伤害。还有困难成功，极难成功，大成功等的效果加成。具体效果由你决定。
      
      重要：当你需要掷骰子时，必须使用roll_dice工具，伤害也要通过roll_dice工具计算。
      
      现在正在进行战斗轮，玩家输入的行动需要进行合法性判断。
      玩家输入的行动: {input}

      如果行为合法，需要把玩家输入的行为造成的结果完全描述出来放进description里。比如：玩家对怪物使用了武器，需要描述武器的命中，怪物的闪避或者对抗，怪物的血量变化，怪物的状态变化，怪物的死亡，等等。！！不要忘了带上掷骰子的动作和结果。
      如果行为不合法，需要把不合法的原因放进description里。
      如果玩家的行为造成了数值变化或者location变化，需要把把更新后的对应participant对象放进result数组里。如玩家对食尸鬼造成1点伤害，那么result数组里需要有食尸鬼的更新后的对象，hp比之前少1点。
      请分析玩家输入并返回结构化结果：
      {{
        isValid: 输入是否合法,
        description: 不合法的原因，或者合法的行动信息(具体做了什么，造成了什么影响，),
        result: participants中发生数据变化的对象[],
    }}
      agent_scratchpad: {agent_scratchpad}
      `);
    const agent = await createToolCallingAgent({
      llm,
      tools,
      prompt,
    });

    const agentExecutor = new AgentExecutor({
      maxIterations: 5, // 最大迭代次数
      agent,
      tools,
    });

    const { output } = await agentExecutor.invoke({
      input: "玩家使用手枪攻击食尸鬼",
    });
    console.log("✅ API 连接成功！");
    console.log("回复:", output);
    
  } catch (error) {
    console.error("❌ API 连接失败:");
    console.error("错误类型:", error.constructor.name);
    console.error("错误信息:", error.message);
    
    if (error.message.includes("fetch failed")) {
      console.log("\n💡 可能的解决方案:");
      console.log("1. 检查网络连接");
      console.log("2. 检查 API 密钥是否正确");
      console.log("3. 检查 API 配额是否用完");
      console.log("4. 尝试使用代理或 VPN");
    }
  }
}

testAPI(); 

File: test_api.py
# === test_api.py ===

import os
import asyncio
from dotenv import load_dotenv
from langchain_google_genai import ChatGoogleGenerativeAI
from src.tools.dice_tools import roll_dice_tool, roll_dice

# 加载环境变量
load_dotenv()

async def test_gemini_connection():
    """测试Gemini API连接"""
    print("🔍 测试Gemini API连接...")
    
    try:
        llm = ChatGoogleGenerativeAI(
            model=os.getenv("GEMINI_MODEL", "gemini-2.0-flash"),
            google_api_key=os.getenv("GOOGLE_API_KEY"),
            temperature=0.1,
        )
        
        response = await llm.ainvoke("你好，请简单回复'连接成功'")
        print(f"✅ Gemini API连接成功: {response.content}")
        return True
        
    except Exception as e:
        print(f"❌ Gemini API连接失败: {e}")
        return False

def test_dice_system():
    """测试骰子系统"""
    print("\n🎲 测试骰子系统...")
    
    test_cases = [
        "1d20",
        "2d6+3", 
        "1d100-5",
        "3d4",
        "1d10"
    ]
    
    for dice in test_cases:
        try:
            result = roll_dice(dice)
            print(f"✅ {dice}: {result}")
        except Exception as e:
            print(f"❌ {dice}: 错误 - {e}")

async def test_dice_tool():
    """测试骰子工具"""
    print("\n🔧 测试骰子工具...")
    
    try:
        result = await roll_dice_tool.ainvoke({"dice_notation": "2d6+3"})
        print(f"✅ 骰子工具测试成功: {result}")
        return True
    except Exception as e:
        print(f"❌ 骰子工具测试失败: {e}")
        return False

async def test_agents():
    """测试智能体系统"""
    print("\n🤖 测试智能体系统...")
    
    try:
        from classes import GraphState, Participant, ParticipantStatus
        from src.agents import player_input_triage_agent
        
        # 创建测试状态
        test_participant = Participant(
            id="test_player",
            name="测试玩家",
            type="investigator",
            stats={"HP": 10, "DEX": 50},
            status=ParticipantStatus.ACTIVE,
            effects=[],
            items=[]
        )
        
        test_state = GraphState(
            participants=[test_participant],
            initiative_order=["test_player"],
            current_actor_index=0,
            player_input="我使用手枪攻击"
        )
        
        # 测试意图分类
        result = player_input_triage_agent(test_state)
        print(f"✅ 意图分类测试成功: {result}")
        return True
        
    except Exception as e:
        print(f"❌ 智能体测试失败: {e}")
        return False

async def test_workflow():
    """测试工作流"""
    print("\n🔄 测试工作流...")
    
    try:
        from src.coc_keeper import combat_workflow
        from classes import GraphState, Participant, ParticipantStatus
        
        # 创建测试状态
        test_participant = Participant(
            id="test_player",
            name="测试玩家",
            type="investigator",
            stats={"HP": 10, "DEX": 50},
            status=ParticipantStatus.ACTIVE,
            effects=[],
            items=[]
        )
        
        test_state = GraphState(
            participants=[test_participant],
            initiative_order=["test_player"],
            current_actor_index=0,
            player_input="我使用手枪攻击"
        )
        
        config = {
            "configurable": {
                "thread_id": "test_workflow"
            }
        }
        
        # 运行工作流
        result = await combat_workflow.ainvoke(test_state, config)
        print(f"✅ 工作流测试成功: 状态更新完成")
        return True
        
    except Exception as e:
        print(f"❌ 工作流测试失败: {e}")
        return False

async def main():
    """主测试函数"""
    print("🧪 开始API和系统测试...")
    print("=" * 50)
    
    # 检查环境变量
    api_key = os.getenv("GOOGLE_API_KEY")
    if not api_key:
        print("❌ 未找到GOOGLE_API_KEY环境变量")
        print("请确保在.env文件中设置了正确的API密钥")
        return
    
    # 运行测试
    tests = [
        ("Gemini API连接", test_gemini_connection),
        ("骰子系统", lambda: test_dice_system()),
        ("骰子工具", test_dice_tool),
        ("智能体系统", test_agents),
        ("工作流", test_workflow)
    ]
    
    results = []
    for test_name, test_func in tests:
        try:
            if asyncio.iscoroutinefunction(test_func):
                result = await test_func()
            else:
                result = test_func()
            results.append((test_name, result))
        except Exception as e:
            print(f"❌ {test_name}测试出错: {e}")
            results.append((test_name, False))
    
    # 输出测试结果
    print("\n" + "=" * 50)
    print("📊 测试结果汇总:")
    
    passed = 0
    total = len(results)
    
    for test_name, result in results:
        status = "✅ 通过" if result else "❌ 失败"
        print(f"{test_name}: {status}")
        if result:
            passed += 1
    
    print(f"\n总计: {passed}/{total} 个测试通过")
    
    if passed == total:
        print("🎉 所有测试通过！系统运行正常。")
    else:
        print("⚠️ 部分测试失败，请检查配置和依赖。")

if __name__ == "__main__":
    asyncio.run(main()) 